src/test/scala/chiselTests/experimental/verification/VerificationSpec.scala

```diff
@@ -20,25 +24,81 @@ class VerificationModule extends Module {
   }
 }
 
-class VerificationSpec extends ChiselPropSpec {
+/** The following test marks verification components with dummy annotations. */
+object VerificationAnnotationTest {
+  /** Dummy verification annotation.
+    * @param target target of component to be annotated
+    */
+  case class VerifAnnotation(target: ReferenceTarget) extends SingleTargetAnnotation[ReferenceTarget] {
+    def duplicate(n: ReferenceTarget): VerifAnnotation = this.copy(target = n)
+  }
+
+  object VerifAnnotation {
+    /** Create annotation for a given verification component.
+      * @param c component to be annotated
+      */
+    def annotate(c: formal.BaseSim): Unit = {
+      chisel3.experimental.annotate(new ChiselAnnotation {
+        def toFirrtl: VerifAnnotation = VerifAnnotation(c.toTarget)
+      })
+    }
+  }
+
+  /** Circuit that contains and annotates verification components. */
+  class AnnotationTest extends Module {
+    val io = IO(new Bundle{
+      val in = Input(UInt(8.W))
+      val out = Output(UInt(8.W))
+    })
+    io.out := io.in
+    val cov = formal.cover(io.in === 3.U)
+    VerifAnnotation.annotate(cov)
+    val assm = formal.assume(io.in =/= 2.U)
+    VerifAnnotation.annotate(assm)
+    val asst = formal.assert(io.out === io.in)
+    VerifAnnotation.annotate(asst)
+  }
+}
 
-  def assertContains[T](s: Seq[T], x: T): Unit = {
-    val containsLine = s.map(_ == x).reduce(_ || _)
+class VerificationSpec extends ChiselPropSpec with Matchers {
+
+  def assertContains(s: Seq[String], x: String): Unit = {
+    val containsLine = s.map(_.contains(x)).reduce(_ || _)
     assert(containsLine, s"\n  $x\nwas not found in`\n  ${s.mkString("\n  ")}``")
   }
 
   property("basic equality check should work") {
-    val fir = ChiselStage.emitChirrtl(new VerificationModule)
+    val fir = ChiselStage.emitChirrtl(new SimpleTest)
     val lines = fir.split("\n").map(_.trim)
 
     // reset guard around the verification statement
-    assertContains(lines, "when _T_2 : @[VerificationSpec.scala 16:15]")
-    assertContains(lines, "cover(clock, _T, UInt<1>(\"h1\"), \"\") @[VerificationSpec.scala 16:15]")
+    assertContains(lines, "when _T_2 : @[VerificationSpec.scala")
+    assertContains(lines, "cover(clock, _T, UInt<1>(\"h1\"), \"\") @[VerificationSpec.scala")
+
+    assertContains(lines, "when _T_6 : @[VerificationSpec.scala")
+    assertContains(lines, "assume(clock, _T_4, UInt<1>(\"h1\"), \"\") @[VerificationSpec.scala")
+
+    assertContains(lines, "when _T_9 : @[VerificationSpec.scala")
+    assertContains(lines, "assert(clock, _T_7, UInt<1>(\"h1\"), \"\") @[VerificationSpec.scala")
+  }
+
+  property("annotation of verification constructs should work") {
+    val testDir = new File("test_run_dir", "VerificationAnnotationTests")
+    (new ChiselStage).emitFirrtl(
+      gen = new VerificationAnnotationTest.AnnotationTest,
+      args = Array ("-td", testDir.getPath)
+    )
```

Comment:
I think this test should check that the verification nodes have the right names in the FIRRTL

Reply:
oh, i didn't even update the emitter to emit the names. Are there guidelines I should follow for emission?

Should I reuse `node`?
```
node assm = assume(clock, _T_3, UInt<1>(1), "") @[VerificationSpec.scala 56:29]
```

or make something new?
```
verif assm : assume(clock, _T_3, UInt<1>(1), "") @[VerificationSpec.scala 56:29]
```

Should I also do something special on the SV emission with the names? Like turn them into labels?

Reply:
The syntax for named statements in firrtl looks like this:

```
assume(clock, _T_3, UInt<1>(1), "") : assm @[VerificationSpec.scala 56:29]
```

Reply:
Thanks Kevin! I guess messing with whether or not the name comes out as a label has to be done in FIRRTL...

Reply:
If you provide a non-empty name (i.e. not `""`), `firrtl` will always include it when serializing.
I am not familiar with the `chisel` code base, but my understanding was that it has its own Chirrtl serializer, no?